src/Symfony/Component/Messenger/TraceableMessageBus.php

```diff
@@ -29,9 +29,10 @@ public function __construct(MessageBusInterface $decoratedBus)
     /**
      * {@inheritdoc}
      */
-    public function dispatch($message): Envelope
+    public function dispatch($message, array $stamps = []): Envelope
     {
         $envelope = $message instanceof Envelope ? $message : new Envelope($message);
+        $envelope = $envelope->with(...$stamps);
```

Comment:
That's where the `Envelope::wrap($messageOrEnvelope, array $stamps)` method would be convenient so we don't need to duplicate these two lines across files.